File: db4onqopt/test/src/com/db4o/test/nativequery/cats/SodaCatPredicate.java
/* Copyright (C) 2004 - 2005  db4objects Inc.  http://www.db4o.com */

package com.db4o.test.nativequery.cats;

import com.db4o.*;
import com.db4o.query.*;


public abstract class SodaCatPredicate extends Predicate {
    
    public void sodaQuery(ObjectContainer oc){
        Query q = oc.query();
        q.constrain(Cat.class);
        constrain(q);
        q.execute();
    }
    
    public abstract void constrain(Query q);
    

}

File: db4on/test/src/com/db4o/test/nativequeries/cats/SodaCatPredicate.cs

using com.db4o;

namespace com.db4o.test.nativequeries.cats
{
	public abstract class SodaCatPredicate : com.db4o.query.Predicate
	{
		private int _count;

		public virtual void SodaQuery(ObjectContainer oc)
		{
			com.db4o.query.Query q = oc.Query();
			q.Constrain(typeof(Cat));
			Constrain(q);
			q.Execute();
		}

		public abstract void Constrain(com.db4o.query.Query q);

#if NET_2_0
        public abstract void DelegateNQ(ObjectContainer oc);
#endif

		public virtual void SetCount(int count)
		{
			_count = count;
		}

		public virtual int Lower()
		{
			return _count / 2 - 1;
		}

		public virtual int Upper()
		{
			return _count / 2 + 1;
		}

	}
}
